modules/learn/pages/clusters-and-availability/xdcr-with-scopes-and-collections.adoc

```diff
@@ -0,0 +1,310 @@
+
+[#migration]
+== Migration
+
+When a pre-7.0 version of Couchbase Server is upgraded to 7.0 or later, all documents that resided in a pre-7.0 bucket appear in the upgraded bucket's _default collection_, within its _default scope_.
+See xref:learn:data/scopes-and-collections.adoc[Scopes and Collections], for information.
+
+Following upgrade, data within the default collection can be _migrated_ to administrator-defined collections, within new target buckets, potentially on the same cluster.
+For each new collection, a replication to the appropriate target bucket can be defined, and a filter applied, ensuring that only the appropriate subset of documents is replicated.
+The mapping between the documents currently in the default collection on the source and the new collection on the target is therefore _explicitly_ specified by the administrator.
+
+Migration, which is only available in Couchbase Server Enterprise Edition, can be illustrated by the following diagram:
+
+image::clusters-and-availability/xdcr-collections-migration-diagram.png[,540,align=left]
+
+The annotations are as follows:
+
+. The administrator must explicitly specify a target and a source bucket, between which replication is to occur.
+
+. The administrator must explicitly specify a target scope and collection, within the target bucket.
+Here, the target scope is `US-Scope`, within which resides the target collection, `Airline-Collection`.
+The depicted goal is to migrate all documents that correspond to US airlines to the target collection: therefore, the administrator must specify a filter such as the following: `type == "airline" && country == "United States"`.
+Thus, every document whose `type` is `"airline"`, and whose `country` is `"United States"` is migrated.
+
+. Similarly, to migrate all documents that correspond to UK airports to the target collection `Airport-Collection`, within the scope `UK-Scope`, a filter such as the following is required: `type == "airport" && country == United Kingdom"`.
+
+For the practical steps, see xref:manage:manage-xdcr/replicate-using-scopes-and-collections.adoc#migrate-data-to-a-collection[Migrate Data to a Collection].
+
+[#rules-for-migration]
+=== Rules for Migration
+
+Each XDCR Migration is performed according to an administrator-specified rule.
+The rule must be expressed as a  key-value pair:
+
+* If _all_ the documents in the `&#95;default` collection of the source bucket are to be migrated to the specified target collection, the keyspace `_default._default` must be the key, and the destination keyspace must be the value.
+For example, the rule `{_default._default: California.SanFrancisco}` specifies that all documents in the `_default` collection of the source bucket should be migrated to the `SanFrancisco` collection, within the `California` scope, on the target bucket.
```

Comment:
Just as a follow up since our last discussion with our PM...
We have decided that if such a rule is present (default.default : California.SanFrancisco), then it means that no other rule can exist. In other words, a rule such as (default.default : NewYork:NewYork) is not allowed if the SanFrancisco one is present, nor is a rule that has a filter expression.

Reply:
I've added: Note that for a given replication, only one migration-rule can be used to migrate data from the _default collection on the source. If a second migration-rule, specifying a different target collection, attempts to migrate data from the same _default collection as does the first migration-rule, an error is generated. (This is the case regardless of whether a filter expression is specified for the second migration-rule.) 

(Do I have this right?)

Reply:
That's correct! Thanks.